inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/config/InspectitEnvironment.java

```diff
@@ -0,0 +1,136 @@
+     * After the update has been executed the environment automatically reloads the {@link #currentConfig}
+     * and fires a {@link InspectitConfigChangedEvent} if the configuration changed.
+     *
+     * @param propertiesUpdater the function to execute for updating the properties. It is guaranteed that
+     *                          neither the property sources nor {@link #currentConfig} will change during the execution of propertiesUpdater.
+     */
+    public synchronized void updatePropertySources(Consumer<MutablePropertySources> propertiesUpdater) {
+        propertiesUpdater.accept(getPropertySources());
+        InspectitConfig oldConfig = currentConfig;
+        reloadConfig();
+        if (!currentConfig.equals(oldConfig)) {
+            val event = new InspectitConfigChangedEvent(this, oldConfig, currentConfig);
+            eventDrain.publishEvent(event);
+        }
+    }
+
+    /**
+     * Allows a thread-safe READ-ONLY access to the property sources of the environment.
+     *
+     * @param propertiesAccessor the function to execute for updating the properties. It is guaranteed that
+     *                           neither the property sources nor {@link #currentConfig} will change during the execution of propertiesAccessor.
+     */
+    public synchronized void readPropertySources(Consumer<PropertySources> propertiesAccessor) {
+        propertiesAccessor.accept(getPropertySources());
+    }
+
+    private void reloadConfig() {
+        currentConfig = Binder.get(this).bind("inspectit", InspectitConfig.class).get();
+    }
+
+    @Override
+    protected void customizePropertySources(MutablePropertySources propsList) {
+        //include the standard sources (e.g. environment variables and properties)
+        super.customizePropertySources(propsList);
+
+        addAgentDefaultYaml(propsList);
+
+        reloadConfig();
+        for (val initializer : configurationInitializationSteps) {
+            initializer.accept(propsList, currentConfig);
+            reloadConfig();
+        }
+
+        log.info("Registered Configuration Sources:");
+        getPropertySources().stream().forEach(ps -> log.info("  {}", ps.getName()));
+    }
+
+    private static void addAgentDefaultYaml(MutablePropertySources propsList) {
+        ClassPathResource defaultYamlResource = new ClassPathResource(DEFAULT_CONFIG_PATH, InspectitEnvironment.class.getClassLoader());
+        Properties defaultProps = PropertyFileUtils.readYamlFiles(defaultYamlResource);
+        propsList.addLast(new PropertiesPropertySource(DEFAULT_CONFIG_PROPERTYSOURCE_NAME, defaultProps));
+    }
+
+    private static void addFileBasedConfiguration(MutablePropertySources propsList, InspectitConfig currentConfig) {
+        String path = currentConfig.getConfig().getFileBased().getPath();
+        Path dirPath = Paths.get(path);
+        Boolean enabled = currentConfig.getConfig().getFileBased().isEnabled();
```

Comment:
Is this really an object Boolean? Can this ever be `null` or what's the deal here?